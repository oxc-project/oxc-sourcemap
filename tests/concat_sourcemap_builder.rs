use std::fs;

use oxc_sourcemap::{ConcatSourceMapBuilder, SourceMap, SourcemapVisualizer};

#[test]
fn concat_sourcemap_builder_basic() {
    let dir = std::path::Path::new(file!())
        .parent()
        .unwrap()
        .join("fixtures_concat_sourcemap_builder/basic");

    let mut builder = ConcatSourceMapBuilder::default();
    let mut source = String::new();
    {
        let js = fs::read_to_string(dir.join("dep1.js")).unwrap();
        let js_map = fs::read_to_string(dir.join("dep1.js.map")).unwrap();
        let sourcemap = SourceMap::from_json_string(&js_map).unwrap();
        builder.add_sourcemap(&sourcemap, source.lines().count() as u32);
        source.push_str(&js);
    }
    {
        let js = fs::read_to_string(dir.join("dep2.js")).unwrap();
        let js_map = fs::read_to_string(dir.join("dep2.js.map")).unwrap();
        let sourcemap = SourceMap::from_json_string(&js_map).unwrap();
        builder.add_sourcemap(&sourcemap, source.lines().count() as u32);
        source.push_str(&js);
    }
    {
        let js = fs::read_to_string(dir.join("dep3.js")).unwrap();
        let js_map = fs::read_to_string(dir.join("dep3.js.map")).unwrap();
        let sourcemap = SourceMap::from_json_string(&js_map).unwrap();
        builder.add_sourcemap(&sourcemap, source.lines().count() as u32);
        source.push_str(&js);
    }

    let sourcemap = builder.into_sourcemap();
    // encode and decode to test token chunk serialization
    let sourcemap = SourceMap::from_json(sourcemap.to_json()).unwrap();
    let visualizer = SourcemapVisualizer::new(&source, &sourcemap);
    let visualizer_text = visualizer.into_visualizer_text();
    insta::assert_snapshot!("concat_sourcemap_builder_basic", visualizer_text);
}
