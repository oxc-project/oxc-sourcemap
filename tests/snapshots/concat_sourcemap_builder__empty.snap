---
source: tests/concat_sourcemap_builder.rs
expression: visualizer_text
snapshot_kind: text
---
- ../src/dep1.js
(0:0) "console." --> (2:0) "console."
(0:8) "log(" --> (2:8) "log("
(0:12) "'dep1')" --> (2:12) "\"dep1\")"
(0:19) ";\n" --> (2:19) ";\n"
- ../src/dep3.js
(0:0) "console." --> (12:0) "console."
(0:8) "log(" --> (12:8) "log("
(0:12) "'dep3')" --> (12:12) "\"dep3\")"
(0:19) ";\n" --> (12:19) ";\n"
